src/main/java/Reporter.java

```diff
@@ -27,65 +28,99 @@
-        JsonObject json1 = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/rt;URL")
+        JsonObject json = new JsonObject().put("id", "MYRT");
+        req.end(json.encode());
+      }, fut1);
+
+      fut1.compose(Void -> {
+        // step 2: create a vertx resource
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created vertx resource!");
+            fut2.complete();
+          } else {
+            fut2.fail("Fail when creating resource vertx");
+          }
+        });
+        JsonObject json = new JsonObject().put("id", vertxResourceId).put("resourceTypePath", "/f;" + feedId + "/rt;MYRT")
                 .put("properties", new JsonObject().put("type", "standalone"));
-        request1.end(json1.encode());
+        req.end(json.encode());
       }, fut2);
 
-      fut2.compose(response -> {
+      fut2.compose(Void -> {
         // step 3: create a event bus resource
-        HttpClientRequest request2 = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(response2 -> {
-          System.out.println("Created event_bus resource!");
-          fut3.complete();
+        HttpClientRequest req = reporter.getHttpRequest(host, resourceURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created event_bus resource!");
+            fut3.complete();
+          } else {
+            fut3.fail("Fail when creating resource event bus");
+          }
         });
-        JsonObject json2 = new JsonObject().put("id", eventBusResourceId).put("resourceTypePath", "/rt;URL");
-        request2.end(json2.encode());
+        JsonObject json = new JsonObject().put("id", eventBusResourceId).put("resourceTypePath", "/f;" + feedId + "/rt;MYRT");
+        req.end(json.encode());
       }, fut3);
 
-      fut3.compose(response -> {
+      fut3.compose(Void -> {
         // step 4: create a metric type
-        HttpClientRequest request3 = reporter.getHttpRequest(host, metricTypeURI, HttpMethod.POST).handler(response3 -> {
-          System.out.println("Created metric type COUNTER!");
-          fut4.complete();
+        HttpClientRequest req = reporter.getHttpRequest(host, metricTypeURI, HttpMethod.POST).handler(resp -> {
+          if (resp.statusCode() == 201) {
+            System.out.println("Created metric type COUNTER!");
+            fut4.complete();
+          } else {
+            fut4.fail("Fail when metric type COUNTER");
+          }
         });
-        JsonObject json3 = new JsonObject().put("id", metricTypeId).put("type", "COUNTER").put("unit", "NONE").put("collectionInterval", "30");
-        request3.end(json3.encode());
+        JsonObject json = new JsonObject().put("id", metricTypeId).put("type", "COUNTER").put("unit", "NONE").put("collectionInterval", "30");
+        req.end(json.encode());
```

Comment:
I believe so.

We will eventually present this to Lukas to get his feeling about our 
usage of Inventory. Please write down all your assumptions and we'll get 
his expert advice later on.

Le 12/07/2016 à 03:42, Austin Kuo a écrit :
> In src/main/java/Reporter.java
> <https://github.com/auszone/vertxFeedReporter/pull/1#discussion_r70366239>:
>
>>          });
>> -        JsonObject json3 = new JsonObject().put("id", metricTypeId).put("type", "COUNTER").put("unit", "NONE").put("collectionInterval", "30");
>> -        request3.end(json3.encode());
>> +        JsonObject json = new JsonObject().put("id", metricTypeId).put("type", "COUNTER").put("unit", "NONE").put("collectionInterval", "30");
>> +        req.end(json.encode());
>
> The metric type is not incorporated with any resource type yet.
> Shall I make 2 different resource type, one for event-bus and one for vertx.
> And then make this metric type incorporated with event-bus resource type?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/auszone/vertxFeedReporter/pull/1/files/d5c2088a0e5f3f5ee8e9f043621385a203723714#r70366239>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABbltvC0V_vptgpqmhZKqQo0Tq9PVBP3ks5qUvDzgaJpZM4JIN0j>.
>